ui/ui.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2015 Square Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ui
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/square/metrics/api"
+	"github.com/square/metrics/query"
+)
+
+type QueryHandler struct {
+	API     api.API
+	Backend api.Backend
+}
+
+type Response struct {
+	Success bool        `json:"success"`
+	Message string      `json:"message,omitempty"`
+	Body    interface{} `json:"body,omitempty"`
+}
+
+func errorResponse(writer http.ResponseWriter, code int, err error) {
+	writer.WriteHeader(code)
+	encoded, err := json.MarshalIndent(Response{Success: false, Message: err.Error()}, "", "  ")
+	if err != nil {
+		writer.Write([]byte("{success:false, message:'failed to encode error message'}"))
```

Comment:
let's return a correct JSON string (the key-values must be quoted with double quotes)